packages/atlas/src/api/queries/storage.graphql

```diff
@@ -43,3 +43,17 @@ query GetBasicStorageBuckets {
     id
   }
 }
+
+query GetStorageBucketsNodeEndpointForBag($bagId: ID) {
+  storageBuckets(
+    where: {
+      operatorStatus_json: { isTypeOf_eq: "StorageBucketOperatorStatusActive" }
+      bags_some: { id_eq: $bagId }
+      operatorMetadata: { nodeEndpoint_contains: "http" }
+    }
+  ) {
+    operatorMetadata {
+      nodeEndpoint
+    }
+  }
+}
```

Comment:
This data shouldn't be fetched from storage provider, but from a distribution provider. Seems no new queries should be needed, we should already have bag->provider mapping with out current assets provider